pkg/istio/authorizationpolicy/controller.go

```
@@ -0,0 +1,463 @@
+			// it can just pass in authz policy name as arg to func convertAthenzModelIntoIstioAuthzPolicy
+			label := obj.Spec.Selector.MatchLabels["svc"]
+			domainRBAC := m.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain, &c.AdIndexInformer)
+			convertedCR := c.RbacProviderV2.ConvertAthenzModelIntoIstioAuthzPolicy(domainRBAC, obj.Namespace, obj.Name, label)
+			if !c.Dryrun {
+				// prevent manual editing the file
+				if castItem.Operation == model.EventUpdate {
+					// assign current revision, update function requires a defined resource version
+					convertedCR.ResourceVersion = obj.ResourceVersion
+					_, err := c.ConfigStoreCache.Update(convertedCR)
+					if err != nil {
+						log.Errorln("error updating authz policy: ", err.Error())
+						return err
+					}
+				}
+				// prevent
+				if castItem.Operation == model.EventDelete {
+					_, err := c.ConfigStoreCache.Create(convertedCR)
+					if err != nil {
+						log.Errorln("error updating authz policy: ", err.Error())
+						return err
+					}
+				}
+			} else {
+				err := c.createDryrunResource(convertedCR, obj.Name, obj.Namespace)
+				if err != nil {
+					return fmt.Errorf("unable write to file, err: %v", err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// checkAuthzEnabledAnnotation checks if current servce object has "authz.istio.io/enabled" annotation set
+func (c *Controller) checkAuthzEnabledAnnotation(serviceObj *corev1.Service) bool {
+	if _, ok := serviceObj.Annotations[authzEnabledAnnotation]; ok {
+		if serviceObj.Annotations[authzEnabledAnnotation] == authzEnabled {
+			return true
+		} else {
+			return false
+		}
+	}
+	return false
+}
+
+func (c *Controller) createDryrunResource(convertedCR model.Config, authzPolicyName string, namespace string) error {
+	convertedObj, err := crd.ConvertConfig(collections.IstioSecurityV1Beta1Authorizationpolicies, convertedCR)
+	if err != nil {
+		log.Errorln("Unable to convert authorization policy config to istio objects")
```

Shouldn't we return the error here?